lib/broadway_kafka/producer.ex

```diff
@@ -215,6 +216,19 @@ defmodule BroadwayKafka.Producer do
     * `[:broadway_kafka, :assignments_revoked, :start | :stop | :exception]` spans -
       these events are emitted in "span style" when receiving assignments revoked call from consumer group coordinator
       See `:telemetry.span/3`.
+
+  ## Shared Client Performance
+
+  Enable shared client may drastically decrease performance. This happens because mutiple producers may block
+  each other waiting for the client response since the connection is hidden inside a process it becomes a
+  bottleneck.
+
+  This is more likely to be an issue if the producers on your pipeline are fetching message from multiple topics
+  and specially if there are very low traffic topics in the mix because of batch wait times.
+
+  In summary to mitigate this you can split your topics between multiple pipelines, but notice that this will
+  increase the resource usage as well creating one new client/connection for each pipeline effectively diminishing
+  the shared_client resource usage gains.
```

Comment:
```suggestion
  To mitigate this, you can split your topics between multiple pipelines, but notice that this will
  increase the resource usage as well. By creating one new client/connection for each pipeline,
  you effectively diminishing the `shared_client` resource usage gains. So make sure to measure
  if you enable this option.
```